src/modules/featureFlag/featureFlag.ts

```diff
@@ -0,0 +1,54 @@
+import Configstore from 'configstore'
+import { join } from 'path'
+import { CLIPreTasks } from '../../CLIPreTasks/CLIPreTasks'
+
+export interface IFeatureFlag {
+  storeFilePath: string
+  getLastFeatureFlagUpdate: () => number
+  setLastFeatureFlagUpdate: (date: number) => void
+  getAllFeatureFlagInfo: () => Record<string, any>
+  getFeatureFlagInfo: <T>(flagName: string) => T
+  setFeatureFlagInfo: (info: Record<string, any>) => void
+}
+
+export class FeatureFlag implements IFeatureFlag {
+  public static readonly FEATURE_FLAG_STORE_FILENAME = 'feature-flag.json'
+  private static singleton: FeatureFlag
+
+  public static getSingleton(): FeatureFlag {
+    if (FeatureFlag.singleton) {
+      return FeatureFlag.singleton
+    }
+
+    FeatureFlag.singleton = new FeatureFlag(
+      join(CLIPreTasks.PRETASKS_LOCAL_DIR, FeatureFlag.FEATURE_FLAG_STORE_FILENAME)
+    )
+    return FeatureFlag.singleton
+  }
+
+  private store: Configstore
+
+  constructor(public storeFilePath: string) {
```

Comment:
Is it private by default?

Reply:
Yes, because the user is going to use only the public methods that access the store.